sdk/python/resources/connections/connections.ipynb

```diff
@@ -0,0 +1,217 @@
+    "ml_client = MLClient(\n",
+    "    DefaultAzureCredential(), subscription_id, resource_group, workspace\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 2 Create Workspace Connection\n",
+    "An Azure Machine Learning workspace is secure and trustworthy way for AML users to store credentials to access external resources from AML\n",
+    "\n",
+    "Workspace Connection can be created using the `WorkspaceConnection` class. This class accepts the following key parameters:\n",
+    "- `name` - Name of connection.\n",
+    "- `target` - The URL or ARM resource ID of the external resource.\n",
+    "- `credentials` - The credentials for authenticating to the external resource.\n",
+    "- `type` - The category of external resource for this connection.\n",
+    "\n",
+    "## 2.1 Create a Workspace Connection\n",
+    "In this example we will create a **Workspace Connection** to an [**Azure Container Registry**]([ACR](https://azure.microsoft.com/en-us/products/container-registry/)) and use **username/password** credentials\n",
+    "\n",
+    "Once connection is successfully created you can reference images from [Azure Container Registry]([ACR](https://azure.microsoft.com/en-us/products/container-registry/)) in AML environment and AML will automatically fetch these credentials while building the environment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.ai.ml.entities import WorkspaceConnection\n",
+    "from azure.ai.ml.entities import UsernamePasswordConfiguration\n",
+    "\n",
+    "credentials = UsernamePasswordConfiguration(\n",
+    "    username=\"<user_name_for_acr>\",\n",
+    "    password=\"<pwd_for_acr>\"\n",
+    ")\n",
+    "\n",
+    "ws_connection = WorkspaceConnection(\n",
+    "    name=\"<connection_name>\",\n",
+    "    target=\"<acr_url>\", # testacr.azurecr.io,\n",
+    "    type=\"container_registry\",\n",
+    "    credentials=credentials\n",
+    ")\n",
+    "\n",
+    "ml_client.connections.create_or_update(ws_connection)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2.2 Create a Workspace Connection for private feeds\n",
+    "In this example we will create a **Workspace Connection** to **private git feed** and use **Personal Access Token (PAT)** credentials\n",
+    "\n",
+    "Once connection is successfully created you can reference feeds from **private git feed** in AML environment and AML will automatically fetch these credentials while building the environment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.ai.ml.entities import WorkspaceConnection\n",
+    "from azure.ai.ml.entities import PatTokenConfiguration\n",
+    "\n",
+    "credentials = PatTokenConfiguration(\n",
+    "    pat=\"<pat_to_git_repo>\"\n",
```

Comment:
We should educate users not to hardcode passwords, but to use environment variables instead.

Reply:
Thanks @rastala  have addressed the comment